src/ui/js/Api.js

```diff
@@ -279,150 +287,92 @@ var Api = (function () {
-      }
-      i += 1;
-    }
-    return true;
-  };
-
   ////////////////////////////////////////////////////////////////////////
   // Load and parse the current player's list of completed games
 
   my.getCompletedGamesData = function(callbackfunc) {
+    my.completed_games = {};
+    var parserargs = [];
+    parserargs.target = my.completed_games;
+    parserargs.isSplit = false;
     my.apiParsePost(
       {'type': 'loadCompletedGames', },
       'completed_games',
-      my.parseCompletedGamesData,
+      my.packageGameData,
       callbackfunc,
-      callbackfunc
+      callbackfunc,
+      parserargs
     );
   };
 
-  my.parseCompletedGamesData = function(data) {
-    my.completed_games.games = [];
-    my.completed_games.nGames = data.gameIdArray.length;
-    var i = 0;
-    while (i < my.completed_games.nGames) {
-      var gameInfo = {
-        'gameId': data.gameIdArray[i],
-        'gameDescription': data.gameDescriptionArray[i],
-        'opponentId': data.opponentIdArray[i],
-        'opponentName': data.opponentNameArray[i],
-        'playerButtonName': data.myButtonNameArray[i],
-        'opponentButtonName': data.opponentButtonNameArray[i],
-        'gameScoreDict': {
-          'W': data.nWinsArray[i],
-          'L': data.nLossesArray[i],
-          'D': data.nDrawsArray[i],
-        },
-        'isAwaitingAction': data.isAwaitingActionArray[i],
-        'maxWins': data.nTargetWinsArray[i],
-        'gameState': data.gameStateArray[i],
-        'status': data.statusArray[i],
-        'inactivity': data.inactivityArray[i],
-        'playerColor': data.playerColorArray[i],
-        'opponentColor': data.opponentColorArray[i],
-      };
-      my.completed_games.games.push(gameInfo);
-      i += 1;
-    }
-    return true;
-  };
-
   ////////////////////////////////////////////////////////////////////////
   // Load and parse the current player's list of rejected games
 
   my.getRejectedGamesData = function(callbackfunc) {
+    my.rejected_games = {};
+    var parserargs = [];
+    parserargs.target = my.rejected_games;
+    parserargs.isSplit = false;
     my.apiParsePost(
       {'type': 'loadRejectedGames', },
       'rejected_games',
-      my.parseRejectedGamesData,
+      my.packageGameData,
       callbackfunc,
-      callbackfunc
+      callbackfunc,
+      parserargs
     );
   };
 
-  my.parseRejectedGamesData = function(data) {
-    my.rejected_games.games = [];
-    my.rejected_games.nGames = data.gameIdArray.length;
-    var i = 0;
-    while (i < my.rejected_games.nGames) {
+  ////////////////////////////////////////////////////////////////////////
+  // Generic parser and repackager of the data returned from the database
+
+  my.packageGameData = function(data, _, parserargs) {
+    // the output format is one of the following:
+    // - split into games awaiting the player and games awaiting the opponent
+    // - made entirely of completed games
```

Comment:
Nit: i would say "- made entirely of games not awaiting action" - they could be completed or rejected (or broken, or some other "not awaiting" category we make up later)